public/app/core/utils/CorsSharedWorker.ts

```diff
@@ -0,0 +1,52 @@
+// Almost identical to CorsWorker.ts. Main difference being it allows loading a SharedWorker if browser supports it
+
+export function sharedWorkersSupported() {
+  return typeof window.SharedWorker !== 'undefined';
+}
+
+class SharedWorkerNotSupported implements SharedWorker {
+  onerror() {}
+  // @ts-ignore
+  readonly port: MessagePort;
+  dispatchEvent(): boolean {
+    return false;
+  }
+  addEventListener(): void {}
+  removeEventListener(): void {}
+}
+
+/**
+ * The base class is created dynamically here to allow later on syntax like:
+ * import { CorsSharedWorker as SharedWorker } from '../utils/CorsSharedWorker';
+ * And also take into account cases where SharedWorked is not available in the browser (mainly mobile browsers)
+ *
+ * It's important to use: new SharedWorker(...) syntax instead of new CorsSharedWorker(...) due to how web-workers
+ * are processing workers syntax (more details https://webpack.js.org/guides/web-workers/)
+ */
+const BaseSharedWorkerClass = sharedWorkersSupported() ? window.SharedWorker : SharedWorkerNotSupported;
```

Comment:
Not sure why we need to return a class with stub like values. How is it going to be used if the shared worker is not supported? Perhaps we can just return null because I'm really not sure how this is going to be handled in the `crashme` repo and it could change in the future leading to unexpected behaviour.

Reply:
Thanks! Yes, I think we can do `class extends null` here instead 👍